### DISCUSSION TITLE
PK-Sim Error: TOO_MUCH_WORK

### DISCUSSION THREAD

**Discussion by Dr-K1997** [Open-Systems-Pharmacology/Forum#1658] [Questions & Problems]
![error](https://github.com/Open-Systems-Pharmacology/Forum/assets/151177172/63e54f52-89b0-443d-9527-9152e082c4ae)

Hi!

I received an error. Description of error as well as PBPK model is as follows: 

Error=Error solving at ODE at time t=195. The solver took 100000 steps but could not reach the output time (TOO_MUCH_WORK). 

PBPK model= It is a mouse model with default properties but changed EHC values in the physiology to account for EHC of the drug. 
The drug has only two metabolites. Events of gallbladder emptying and meal(customized for mice) have been added. Administration is 5 mg/kg.  

Please advise what is it that I am doing wrong. I am a beginner and am not sure how to exactly incorporate enterohepatic circulation for a drug in mice while also accounting for gallbladder emptying and a mouse customized meal. 

IMPORTANT: I just re-ran the simulation after deleting the meal event and it completed which means that the problem is with meal. I am using roughly estimated meal values for a mouse and have attached a snapshot of my values. What should be done in this case?
 
![meal](https://github.com/Open-Systems-Pharmacology/Forum/assets/151177172/eedd7e3d-9843-49f6-9c66-ecd5c36811eb)

Thanking in advance!

**Comment by Yuri05** [comment]
The meal volume is **0.5 L** - seems to be a bit too high for a mouse 😉

**Comment by NingJia012** [comment]
Hello, I've encountered the same error too. I'd like to ask if you have solved this problem now?

**Reply by Dr-K1997** [reply]
yes. Meal volume should correspond to size of stomach of specie of interest.

**Comment by rwmcintosh** [comment]
If you see this `TOO_MUCH_WORK` error it indicates that the system of ODE's created from the model are not solvable by the CVODES solver within the amount of steps defined by MXStep.
![image](https://github.com/user-attachments/assets/2e660b1c-d683-4082-8bd0-912f56a7926f)

100000 steps is a lot of steps, so just check that it's not accidentally set to a small number. If you have the default, then it could help to adjust the tolerances.

![image](https://github.com/user-attachments/assets/1a3482af-a203-41f4-b91d-d866be0e73bd)

**Reply by NingJia012** [reply]
Thank you for your reply. At that time, I checked the value of MXStep, and indeed it was set to the default value. Then I adjusted the tolerances, but the model still reported errors. Finally, when I closed the file, I selected "Save the change", and then this model could run normally. I wonder if this can be considered a bug.

**Reply by rwmcintosh** [reply]
Would you try to use the old default tolerances again to make sure that this was the change that made the simulation able to run?

Maybe like this
1. Set the tolerances to default
2. Exit PK-Sim and save
3. Open the project and run the simulation. Do you get an error?
4. Set the tolerances to the working value and run the simulation without saving. Do you get an error
5. Save the project, exit.
6. Open the project and run. Do you get an error?

The reason to do it in this sequence is to isolate the tolerance change as the only change that is, or is not being propagated to the solver unless the project is saved.

If this is repeatable, then it would be very helpful for a bug report.

**Reply by NingJia012** [reply]
Thank you for your reply. I was using version 11.2 at that time. After encountering the "TOO_MUCH_WORK" error, my adjustment process was roughly as follows:

Firstly, I adjusted the tolerances. However, the model still reported the same error. Then, I restored the tolerances to their default values and continued running the model, but it still reported the same error.

After that, I created a new model with the same settings from the PK-Sim.exe file. At the same time, I didn't close the old file window. But even so, the new file still reported the "TOO_MUCH_WORK" error.

After a while, I closed the file and saved the changes. When I reopened the file, the model could run successfully.

Now I have reinstalled version 11.3. So far, I haven't encountered this bug. Therefore, I'm very sorry that I can't reproduce this error.
